github: add Code of Conduct

miri64 opened the pull request:
This PR provides a first draft for our future code of conduct. I think it's about high time we get one (though I also am convinced that we already followed this rules to our best abilities for the most part up to this) ;-).

miri64 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+affect a person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report
+it by emailing <riot@riot-os.org>. For more details please see our [Reporting
```
Might be a good idea to have a separate mailing list (or at least an alias to one of the existing ones) for that.

kaspar030 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+mixed group of professionals and volunteers from all over the world, working on
+every aspect of our goals - including teaching, steering, and connecting people.
+
+We want diversity to be one of our greatest strength, but it can also lead to
```
s/strength/strengths/

kaspar030 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+This code of conduct applies to all spaces managed by the RIOT community. This
+includes IRC, the mailing lists, our GitHub projects, face to face events, and
+any other forums created by the community for communication within the
+community. In additions, violations of this code outside these spaces may also
```
addition

kaspar030 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+    to members of any race, ethnicity, culture, national origin, colour,
+    immigration status, social and economic class, educational level, sex,
+    sexual orientation, gender identity and expression, age, size, family
+    status, political belief, religion, and mental and physical ability.
```
should we scratch political belief?

miri64 replied:
No, I think it is as important to be in here as e.g. religion. Political beliefs that are not in line with the code of conduct are implicitly exempt anyway, as I read it. ;-)

kaspar030 replied:
ok :)

aabadie commented:
I was also looking at the Django CoC which is itself inspired from [there](http://web.archive.org/web/20141109123859/http://speakup.io/coc.html).

Thanks for providing this !

miri64 commented:
> I was also looking at the Django CoC which is itself inspired from there.

I know, but since I took the Django CoC as a basis and tweaked it a little for our community I was referencing it insteado of the Speak Up! CoC ;-).

aabadie commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+Questions?
+----------
+
+If you have questions, please see [the FAQ](TODO). If that doesn't answer your
```
And we also need a FAQ now :)

miri64 replied:
We [have one](https://github.com/RIOT-OS/RIOT/wiki/FAQ), but this one (at least the linked one by Django) is supposed to be CoC specific.

smlng commented:
@miri64 nice, and thanks for tackling this! I like the text and the adaption made for RIOT. I wouldn't fiddle to much with it, but rather have it in place i.e. published. 

How do you/we want to proceed with the FAQ? IMHO the Django FAQ fit for RIOT, too - though we don't have a foundation or other form of official organisation (yet). But where to put them: wiki or webpage (i.e., riot-os.org)?

miri64 commented:
> But where to put them: wiki or webpage (i.e., riot-os.org)?

I'd say the wiki.

miri64 commented:
Reporting Guidelines need also be created.

miri64 commented:
Since this is something involving the whole community, I'd like to have some more input before we go ahead and merge this.

gebart commented:
I think this is a great addition to the project. 👍

miri64 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,79 @@
+affect a person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report
+it by emailing <riot@riot-os.org>. For more details please see our [Reporting
```
@OlegHahm can we make an alias or something, like conduct@riot-os.org or coc@riot-os.org?

kaspar030 replied:
Not sure I like the sound of the latter in combination with this context.

smlng replied:
in other contexts sometimes `abuse@example.com` is used, otherwise `conduct@riot-os.org` sounds fine, too?!

miri64 replied:
Oh yes, this is another issue, that needs to be addressed before merging this IMHO.

miri64 replied:
> Not sure I like the sound of the latter in combination with this context.

Ooops, only notice this now ^^"

miri64 commented:
Please also discuss the wiki pages I added right now

* [CoC: Reporting Guidelines](https://github.com/RIOT-OS/RIOT/wiki/CoC:-Reporting-Guidelines)
* [CoC: FAQ](https://github.com/RIOT-OS/RIOT/wiki/CoC:-FAQ)

They were adapted from the respective versions by Django as well.

One thing open is: the Django community has the DSF who has a board and also working group that explicitly deals with CoC reports. The latter I replaced with our core maintainers (basically everyone who receives mails to `riot@riot-os.org`), but the first is a little trickier (I marked this with TODO on the page for now). The board basically steps in, if someone is not happy about the decision the CoC-WG made. My initial idea is to maybe add some contact information well trusted members of the RIOT community there, who dealt with such administrative tasks in the past, but maybe someone else has a better idea that we can use before the RIOT association is official ;-).

amcjen commented:
I'm super pleased to see this effort taken seriously and to use existing code language from other successful projects.  I don't have any suggestions, other than the mention to make it clear how to report if something is amiss--but @miri64 already added them to the Wiki pages, so perfect.  👍

smlng commented:
> Please also discuss the wiki pages I added right now
>
> CoC: Reporting Guidelines
> CoC: FAQ
> They were adapted from the respective versions by Django as well.

maybe add a footnote, that we adapted the FAQ from Django?

miri64 commented:
> maybe add a footnote, that we adapted the FAQ from Django?

Done, and also for the reporting guidelines.

smlng commented:
4 approvals in 1 week, looks good - lets see if Murdock agrees too 😉

smlng commented:
well ... squash please.

miri64 commented:
> 4 approvals in 1 week, looks good - lets see if Murdock agrees too :wink:

Well there is still the issue of accountability we need to discuss: Where should people go, if the core team fails to address a report properly in their opinion about CoC violation? Or should we remove this section for now and reinstate this with the board of the association as soon as it is registered?

miri64 commented:
(the section I'm talking about is the last paragraph of the reporting guidelines)

smlng commented:
mhm, so this reconsideration is referring to another _escalation stage_ in the reporting process, right? So if a reporting party does not agree with the initial decision/result of the core community, they might take it to the next step. Which in itself could also fail to agree with the expected outcome of the reporter - so there might always be those cases which are difficult to decide or act upon or come to final conclusion everybody happy with. Further as there is no higher board or association for RIOT (yet) its IMHO fine to remove that paragraph for now - and reconsider as soon as that changes.

miri64 commented:
Yes it is basically about what you would call in legal terms a "second instance". The "third instance" (if that fails) is in case of the RIOT community to leave the community (which depending on the case could be very bad for the community, so the pressure to make the "right" decission for each instance is higher the less instances there are).

miri64 commented:
Ok two issues that are open still:

* What would be our second instance for reporting (or remove the mention of that for now)?
* I can haz abuse@riot-os.org, conduct@riot-os.org or something of the like.

Finally, due to the impact of this on the community I would like to keep this open even with 5 ACKs at least until end of next week.

miri64 commented:
I noticed, that I can create the alias `conduct@riot-os.org` myself and did so. Will adapt the text(s)

miri64 commented:
I removed the section on reconsideration in the reporting guidelines for now. Let's just hope that in the meantime it will never come to that case.

miri64 commented:
Squashed and ready to be merged.

smlng commented:
Let's do this 🎉
